pkg/autodetect/main.go

```diff
@@ -96,12 +97,11 @@ func (b *Background) autoDetectCapabilities() {
 			// the platform won't change during the execution of the operator, need to run it only once
 			b.detectPlatform(ctx, apiList)
 
-			b.detectOAuthProxyImageStream(ctx)
-
 			// the version of the APIs provided by the platform will not change
 			b.detectCronjobsVersion(ctx)
 			b.detectAutoscalingVersion(ctx)
 		})
+		b.detectOAuthProxyImageStream(ctx)
```

Comment:
Since it seems that we need to rerun this multime times, we need to add new field and methods to the `OperatorConfiguration` struct to ensure we don't access `viper` concurrently at the same time.

Reply:
Thanks for that! I've added methods to protect access behind a mutex and moved the image name string into a global constant.